File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int CalculateGrade(string Text);
int CountLetters(string Text);
int CountWords(string Text);
int CountSentences(string Text);

int main(void)
{
    string Texts = get_string("Text: ");
    int GradeIndex = CalculateGrade(Texts);
    if (GradeIndex < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (GradeIndex >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", GradeIndex);
    }
}

int CalculateGrade(string Text)
{
    int TotalWords = CountWords(Text);
    int TotalLetters = CountLetters(Text);
    int TotalSentences = CountSentences(Text);
    float AvgLetters = TotalLetters / (TotalWords / 100.0);
    float AvgSentences = TotalSentences / (TotalWords / 100.0);
    int GradeIndex = round(0.0588 * AvgLetters - 0.296 * AvgSentences - 15.8);
    return GradeIndex;
}

int CountLetters(string Text)
{
    int Counter = 0;
    int TextLength = strlen(Text);
    for (int i = 0; i < TextLength; i++)
    {
        if (isalpha(Text[i]))
        {
            Counter++;
        }
    }
    return Counter;
}
int CountWords(string Text)
{
    int Counter = 0;
    int TextLength = strlen(Text);
    for (int i = 0; i < TextLength; i++)
    {
        if (isspace(Text[i]))
        {
            Counter++;
        }
    }
    return Counter + 1;
}
int CountSentences(string Text)
{
    int Counter = 0;
    int TextLength = strlen(Text);
    for (int i = 0; i < TextLength; i++)
    {
        if (Text[i] == '.' || Text[i] == '!' || Text[i] == '?')
        {
            Counter++;
        }
    }
    return Counter;
}

File: sentimental-readability/readability.py

def main():
    Texts = input("Text: ")
    GradeIndex = CalculateGrade(Texts)
    if GradeIndex < 1:
        print("Before Grade 1")
    elif GradeIndex >= 16:
        print("Grade 16+")
    else:
        print("Grade",GradeIndex)


def CalculateGrade(Text):
    TotalWords = Text.count(" ") + 1
    TotalLetters = 0
    for Letter in Text:
        if Letter.isalpha():
            TotalLetters+=1
    TotalSentences = Text.count(".") + Text.count("!") + Text.count("?")
    AvgLetters = TotalLetters / (TotalWords / 100.0)
    AvgSentences = TotalSentences / (TotalWords / 100.0)
    GradeIndex = int(0.0588 * AvgLetters - 0.296 * AvgSentences - 15.8 + 0.5)
    return GradeIndex


main()
